pkg/apis/core/validation/cloudprofile.go

```diff
@@ -32,6 +32,13 @@ import (
 	kubernetescorevalidation "github.com/gardener/gardener/pkg/utils/validation/kubernetes/core"
 )
 
+var (
+	availableWorkerCRINamesForCloudProfile = sets.New(
+		string(core.CRINameDocker),
```

Comment:
If not, a TODO statement should be added for future removal and the respective condition.

Reply:
> Could we remove the core.CRINameDocker constant and all its occurrences?

No, we still need it in the cloudprofile.

> If not, a TODO statement should be added for future removal and the respective condition.

It should be done more as part of https://github.com/gardener/gardener/issues/4673, right?

Reply:
I think #4673 is only about internal usage of the `docker` binary on shoot nodes. The `core.CRINameDocker` is used for `CloudProfile`s and machine CRI configuration. So I think we can only drop it when we only support Kubernetes versions that do not support Docker anymore. Isn't 1.28 the first version where this applies? This would mean that we can drop it as soon as 1.28 is the least supported version. Is this correct?

Reply:
> Isn't 1.28 the first version where this applies?

I didn't get you. For shoots, we don't support docker for k8s 1.23+. For machines, from https://github.com/gardener/gardener/blob/4a0933893af195eb50cf537ccbfc12a119fdea7b/pkg/apis/core/validation/cloudprofile.go#L310-L312 and the above issue,
> Why is this needed:
While it is possible for users to specify containerd to run their workload containers, Gardener still requires the docker daemon to be present and working in the OS images used for the worker nodes.

It looks like docker is always required as cri.

Reply:
OK, why do we still need `core.CRINameDocker` then for the `CloudProfile`? After this PR no Kubernetes version will be supported where we allow `docker` to be configured as CRI, so shouldn't we rather ask people to remove it from their `CloudProfile`s?

Reply:
> why do we still need core.CRINameDocker then for the CloudProfile?

How else do we stop people from adding machine images which do not support docker CRIs?